optuna/multi_objective/samplers/_nsga2.py

```diff
@@ -0,0 +1,278 @@
+            raise ValueError(
+                "`mutation_prob` must be None or a float value within the range [0.0, 1.0]."
+            )
+
+        if not (0.0 <= crossover_prob <= 1.0):
+            raise ValueError("`crossover_prob` must be a float value within the range [0.0, 1.0].")
+
+        if not (0.0 <= swapping_prob <= 1.0):
+            raise ValueError("`swapping_prob` must be a float value within the range [0.0, 1.0].")
+
+        self._population_size = population_size
+        self._mutation_prob = mutation_prob
+        self._crossover_prob = crossover_prob
+        self._swapping_prob = swapping_prob
+        self._random_sampler = multi_objective.samplers.RandomMultiObjectiveSampler(seed=seed)
+        self._rng = np.random.RandomState(seed)
+
+    def infer_relative_search_space(
+        self,
+        study: "multi_objective.study.MultiObjectiveStudy",
+        trial: "multi_objective.trial.FrozenMultiObjectiveTrial",
+    ) -> Dict[str, BaseDistribution]:
+        return {}
+
+    def sample_relative(
+        self,
+        study: "multi_objective.study.MultiObjectiveStudy",
+        trial: "multi_objective.trial.FrozenMultiObjectiveTrial",
+        search_space: Dict[str, BaseDistribution],
+    ) -> Dict[str, Any]:
+        parent_generation, parent_population = self._collect_parent_population(study)
+        trial_id = trial._trial_id
+
+        generation = parent_generation + 1
+        study._storage.set_trial_system_attr(trial_id, _GENERATION_KEY, generation)
+
+        if parent_generation >= 0:
+            p0 = self._select_parent(study, parent_population)
+            if self._rng.rand() < self._crossover_prob:
+                p1 = self._select_parent(
+                    study, [t for t in parent_population if t._trial_id != p0._trial_id]
+                )
+            else:
+                p1 = p0
+
+            study._storage.set_trial_system_attr(
+                trial_id, _PARENTS_KEY, [p0._trial_id, p1._trial_id]
+            )
+
+        return {}
+
+    def sample_independent(
+        self,
+        study: "multi_objective.study.MultiObjectiveStudy",
+        trial: "multi_objective.trial.FrozenMultiObjectiveTrial",
+        param_name: str,
+        param_distribution: BaseDistribution,
+    ) -> Any:
+        if _PARENTS_KEY not in trial.system_attrs:
+            return self._random_sampler.sample_independent(
+                study, trial, param_name, param_distribution
+            )
+
+        p0_id, p1_id = trial.system_attrs[_PARENTS_KEY]
+        p0 = study._storage.get_trial(p0_id)
+        p1 = study._storage.get_trial(p1_id)
+
+        param = p0.params.get(param_name, None)
+        if param is None or self._rng.rand() < self._swapping_prob:
+            param = p1.params.get(param_name, None)
+
+        mutation_prob = (
+            self._mutation_prob if self._mutation_prob is not None else 1.0 / len(p0.params)
```

Comment:
An edge case but this could result in a zero division error. How about resorting to `p1.params` or making sure it's always mutated in such cases?

Reply:
Nice catch! Fixed: https://github.com/optuna/optuna/pull/1163/commits/ed3475f6f47ff86d3c0f40d956827f9ecc023ef6